app/scripts/migrations/048.js

```diff
@@ -0,0 +1,36 @@
+const version = 48
+import { cloneDeep } from 'lodash'
+import { getPlatform } from '../lib/util'
+import { PLATFORM_BRAVE } from '../lib/enums'
+
+/**
+ * The purpose of this migration is to remove the {@code ipfsGateway} setting and default to native Brave ipfs.
+ */
+export default {
+  version,
+  migrate: async function (originalVersionedData) {
+    const versionedData = cloneDeep(originalVersionedData)
+    versionedData.meta.version = version
+    const state = versionedData.data
+    versionedData.data = transformState(state)
+    return versionedData
+  },
+}
+
+function transformState (state) {
+  // if it is Brave Browser and chrome.ipfs flag is available then use native Brave ipfs
+  const isChromeIpfsAvailable = window?.chrome?.ipfs?.getIPFSEnabled
```

Comment:
We might want to just call this top level and then use the value instead of migrating prefs. Thoughts @ryanml?   And just don't use the pref when it's available.